### PR TITLE
More flexible layers

### LINKED ISSUE #2: Add grid map layer for scenario comparison
We're thinking of comparing yields for historical data vs. each scenario with two map layers, both using circles. Right now, there is only one circle layer that visualizes a combination of these variables:

Screenshot 2023-11-08 at 5 43 36 PM

This change would add a second layer where the model/scenario is always `historical`

### LINKED ISSUE #3: Support styling metric by size and color
We are currently able to visualize any metric by color on the map:

Image

...but don't have a way to easily restyle grid dots by size. Could we set this up so it's easier to play with the data and find interesting cross sections?

### PR DESCRIPTION
**Pull request by ebrelsford**
Closes #2 
Closes #3 

This PR refactors the code base a bit. I'm trying to navigate between making it easier to experiment and add new map instances and also account for the likelihood that the final app will have a number of distinct map instances. So some styling change may require code changes but I think things are a lot more flexible now and will be able to accommodate most needs.

Here's the new map selector:

image

basically I'm thinking while we work through map variants we'll keep a bunch of them around and this dropdown will allow use to pick one. These maps also demonstrate various styles implemented currently.

Through the map selector you can see that we now support multiple grid layers, stroke-only grid layers, and dynamic radii.

### CODE REVIEW

File: vacs-map-app/src/App.vue [hunk 1]
```diff
@@ -1,21 +1,71 @@
...
   </main>
 </template>
 
 <script setup>
-import BaseMap from '@/components/BaseMap.vue';
+import { computed, ref } from 'vue';
+import MapContainerColor from '@/components/MapContainerColor.vue';
+import MapContainerColorRadius from '@/components/MapContainerColorRadius.vue';
+import MapContainerNotFilled from '@/components/MapContainerNotFilled.vue';
+import MapContainerNotFilledTwoLayers from '@/components/MapContainerNotFilledTwoLayers.vue';
```
Comment (ebrelsford) [inline vacs-map-app/src/App.vue:18]: If you make a new map, import it here...

File: vacs-map-app/src/App.vue [hunk 2]
```diff
@@ -1,21 +1,71 @@
...
+  {
+    id: 'not-filled',
+    name: 'circles not filled',
+    component: MapContainerNotFilled,
+  },
+  {
+    id: 'not-filled-2',
+    name: 'circles not filled, two layers',
+    component: MapContainerNotFilledTwoLayers,
+  },
```
Comment (ebrelsford) [inline vacs-map-app/src/App.vue:41]: ...then add your new map to the list of available maps.

File: vacs-map-app/src/components/MapContainerColor.vue [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      <GridOverlay
+        id="grid-layer-1"
+        :color-column="selectedColumn"
+        :color-column-extent="selectedColumnExtent"
+        :color-column-quintiles="selectedColumnQuintiles"
+        :color-diverging="false"
+        :sourceId="sourceId"
+        :map="map"
+        :mapReady="mapReady"
+      />
```
Comment (ebrelsford) [inline vacs-map-app/src/components/MapContainerColor.vue:14]: This is the layer we add to the map. I'd love to make this layer as flexible as we can (you can see in other maps that the props can vary a lot). But if we get to a point where we're making very specific changes to the layer that we don't want to include on all layers, we should consider having multiple components instead.
